docs/messaging/messaging-metrics.md

```diff
@@ -0,0 +1,409 @@
+# Semantic Conventions for Messaging Metrics
+
+**Status**: [Experimental][DocumentStatus]
+
+<!-- Re-generate TOC with `markdown-toc --no-first-h1 -i` -->
+
+<!-- toc -->
+
+- [Producer metrics](#producer-metrics)
+  * [Metric: `messaging.publish.duration`](#metric-messagingpublishduration)
+  * [Metric: `messaging.publish.messages`](#metric-messagingpublishmessages)
+- [Consumer metrics](#consumer-metrics)
+  * [Metric: `messaging.receive.duration`](#metric-messagingreceiveduration)
+  * [Metric: `messaging.receive.messages`](#metric-messagingreceivemessages)
+  * [Metric: `messaging.deliver.duration`](#metric-messagingdeliverduration)
+  * [Metric: `messaging.deliver.messages`](#metric-messagingdelivermessages)
+  * [Metric: `messaging.settle.duration`](#metric-messagingsettleduration)
+  * [Metric: `messaging.settle.messages`](#metric-messagingsettlemessages)
+
+<!-- tocstop -->
+
+## Producer metrics
+
+### Metric: `messaging.publish.duration`
+
+This metric is [required][MetricRequired].
+
+When this metric is reported alongside an messaging publish span, the metric value SHOULD be the same as the corresponding span duration.
```

Comment:
Is there any view about how transactions would play into this? Many systems have transactional publish and settlement.

Reply:
This is a great question for messaging  SIG, but out of the scope of this PR.
In Azure systems, transactions don't change the timing of operations, but add additional scope for the broker. While it may be useful to have transaction-oriented metrics, this is not something we currently support in otel and needs a lot of research and prototyping.

Reply:
created an issue to track it https://github.com/open-telemetry/semantic-conventions/issues/264